lib/mongoid/interceptable.rb

```diff
@@ -162,6 +170,46 @@ def _mongoid_run_child_callbacks(kind, children: nil, &block)
       end
```

Comment:
(GitHub won't let me leave a comment on a line of code that's not actually part of the PR, so I'm leaving it here)

A few lines above this, `_mongoid_run_child_callbacks` is invoked, which is fine. However, since we already know that around callbacks for embeds are enabled, we might as well just invoke `_mongoid_run_child_callbacks_with_around`, and avoid adding an extra call to the stack.

Reply:
Good catch, thank you. Fixed.